src/Bounds.cpp

```diff
@@ -861,6 +965,44 @@ class Bounds : public IRVisitor {
                 // If the argument is unbounded on one side, then the max is unbounded.
                 interval.max = Interval::pos_inf;
             }
+        } else if (op->is_intrinsic(Call::absd)) {
+            internal_assert(!t.is_handle());
+            if (t.is_float()) {
+                Expr e = abs(op->args[0] - op->args[1]);
+                e.accept(this);
+            } else {
+                // absd() for int types will always produce a uint result
+                internal_assert(t.is_uint());
+
+                // Recover what the matched types for the args were
+                Expr a = op->args[0];
+                Expr b = op->args[1];
+                match_types(a, b);
```

Comment:
Why is this necessary if the absd function already did it?

Reply:
You are correct, the match_type'd versions of the args are saved in the op.